config/config-api/src/main/java/com/thoughtworks/go/config/materials/git/GitMaterialConfig.java

```diff
@@ -229,6 +214,13 @@ public void setConfigAttributes(Object attributes) {
             String branchName = (String) map.get(BRANCH);
             this.branch = StringUtils.isBlank(branchName) ? DEFAULT_BRANCH : branchName;
         }
+        if (map.containsKey("userName")) {
```

Comment:
This can be moved to `ScmMaterialConfig`

Reply:
Same as https://github.com/gocd/gocd/pull/6130#discussion_r275265469